tests/Forms/ElementalAreaFieldTest.php

```diff
@@ -46,4 +52,78 @@ public function testFieldReturnsCompositeFieldOnReadonlyTransformation()
             'Uses CompositeField for read-only instances'
         );
     }
+
+    public function testSaveInto()
+    {
+        $field = ElementalAreaField::create(
+            'ElementalAreaField',
+            $this->objFromFixture(ElementalArea::class, 'area1'),
+            [TestElement::class => TestElement::create()->getType()]
+        );
+
+        /* @var BaseElement $element */
+        $element1 = $this->objFromFixture(TestElement::class, 'element1');
+        $element2 = $this->objFromFixture(TestElement::class, 'element2');
+        $element2Value = $element2->TestValue;
+        $page = $element1->getPage();
+        $formID = sprintf(ElementalAreaController::FORM_NAME_TEMPLATE, $element1->ID);
+        $fieldName = sprintf(EditFormFactory::FIELD_NAMESPACE_TEMPLATE, $element1->ID, 'TestValue');
+        $elementData = [
+            $formID => [
+                $fieldName => 'Some new value',
+            ],
+        ];
+        $field->setValue($elementData);
+        $field->saveInto($page);
+
+        $newElement = DataObject::get_by_id(TestElement::class, $element1->ID);
+        $this->assertNotNull($newElement);
+        $this->assertEquals('Some new value', $newElement->TestValue);
+
+        $newElement = DataObject::get_by_id(TestElement::class, $element2->ID);
+        $this->assertNotNull($newElement);
+        $this->assertEquals($element2Value, $newElement->TestValue);
+    }
+
+    public function testSaveIntoExtensible()
+    {
+        $element = $this->objFromFixture(TestElement::class, 'element1');
+        $page = $element->getPage();
+        $formID = sprintf(ElementalAreaController::FORM_NAME_TEMPLATE, $element->ID);
+        $fieldName = sprintf(EditFormFactory::FIELD_NAMESPACE_TEMPLATE, $element->ID, 'TestValue');
+
+        // Apply the extension
+        $extensionMock = $this->getMockBuilder(ElementalAreaFieldExtension::class)
+            ->setMethods(['onSaveInto'])
+            ->getMock();
+        $extensionMock->expects($this->once())
```

Comment:
could you add a comment about what this black magic is?  It's testing something though I'm not actually sure what?

Reply:
Test doubles are black magic? Seems like a bit of a stretch given how much we use them. https://github.com/silverstripe/silverstripe-framework/search?q=in%3Afile+-%3EgetMock&unscoped_q=in%3Afile+-%3EgetMock

Reply:
Yeah I have to agree with @unclecheese here. This is normal usage of PHPUnit mocks, and the API is pretty declarative.